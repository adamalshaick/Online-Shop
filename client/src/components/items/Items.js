import React, { Component } from "react";
import PropTypes from "prop-types";
import ItemFeed from "./ItemFeed";
import SelectListGroup from "../common/SelectListGroup";
import fetchItems from "../common/hoc/fetchItems";
import fetchCurrentUser from "../common/hoc/fetchCurrentUser";
import { SecondaryHeader } from "../common/styles/Header";
import {
  sortOptions,
  selectOptions,
  sort,
  select
} from "../../utils/browseItems";

class Items extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sortBy: "",
      selectedCategory: ""
    };
  }

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  render() {
    const { items } = this.props.item;
    const itemsArray = select(this.state.selectedCategory, items);
    sort(this.state.sortBy, itemsArray);

    return (
      <div className="container">
        <div className="row">
          <div className="col-md-12">
            <SecondaryHeader className="m-5 text-center">
              Items for sale
            </SecondaryHeader>
            <div className="row">
              <div className="col-md-6">
                <SelectListGroup
                  placeholder="* Category"
                  name="selectedCategory"
                  value={this.state.selectedCategory}
                  options={selectOptions}
                  onChange={this.onChange}
                />
              </div>
              <div className="col-md-6">
                <SelectListGroup
                  placeholder="* Sort by..."
                  name="sortBy"
                  value={this.state.sortBy}
                  options={sortOptions}
                  onChange={this.onChange}
                />
              </div>
            </div>
            <ItemFeed
              items={itemsArray}
              currentUser={this.props.user.currentUser}
            />
          </div>
        </div>
      </div>
    );
  }
}

Items.propTypes = {
  item: PropTypes.object.isRequired,
  cart: PropTypes.object.isRequired
};

export default fetchCurrentUser(fetchItems(Items));
