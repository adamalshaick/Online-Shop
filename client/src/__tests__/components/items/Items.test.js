import React from "react";
import { Items } from "../../../components/items/Items.js";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const Props = {
  item: { items: [] },
  user: { currentUser: {} }
};

const wrapper = shallow(<Items {...Props} />);
describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
