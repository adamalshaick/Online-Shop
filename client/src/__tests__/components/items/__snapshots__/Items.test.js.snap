// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <styled.header
        className="m-5 text-center"
      >
        Items for sale
      </styled.header>
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <SelectListGroup
            name="selectedCategory"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "label": "* Select category",
                  "value": "",
                },
                Object {
                  "label": "Clothes",
                  "value": "Clothes",
                },
                Object {
                  "label": "Electronics",
                  "value": "Electronics",
                },
                Object {
                  "label": "Shoes",
                  "value": "Shoes",
                },
              ]
            }
            placeholder="* Category"
            value=""
          />
        </div>
        <div
          className="col-md-6"
        >
          <SelectListGroup
            name="sortBy"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "label": "* Sort by",
                  "value": "",
                },
                Object {
                  "label": "Price ascending",
                  "value": "ascending",
                },
                Object {
                  "label": "Price descending",
                  "value": "descending",
                },
              ]
            }
            placeholder="* Sort by..."
            value=""
          />
        </div>
      </div>
      <ItemFeed
        currentUser={Object {}}
        items={Array []}
      />
    </div>
  </div>
</div>
`;
